src/integrationTest/java/uk/gov/hmcts/reform/sendletter/e2e/EndToEndTest.java

```diff
@@ -84,10 +87,21 @@ private boolean letterHasBeenPosted() {
     }
 
     private void createXeroxReport(LocalSftpServer server) throws IOException {
-        Stream<UUID> letterIds = Arrays.stream(server.lettersFolder.list()).map(FileNameHelper::extractId);
+        Stream<UUID> letterIds = Arrays.stream(server.lettersFolder.listFiles()).map(this::validateGettingLetterId);
         XeroxReportWriter.writeReport(letterIds, server.reportFolder);
     }
 
+    // Validate that the file is a zipped PDF,
+    // returning the Letter's ID from the filename.
+    private UUID validateGettingLetterId(File file) {
+        try {
+            PdfHelper.validateZippedPdf(Files.readAllBytes(file.toPath()));
+            return FileNameHelper.extractId(file.getName());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
or maybe just change test singature to `throws Exception`?

Reply:
Can't throw checked exceptions from a method passed to map() unfortunately

Reply:
:love: java...